arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -2069,10 +2069,10 @@ public inline fun <A, B> Either<A, B>.getOrHandle(default: (A) -> B): B =
  */
 @Deprecated(
   RedundantAPI + "Prefer if-else statement inside either DSL, or replace with explicit flatMap",
-  ReplaceWith("flatMap { b -> b.takeIf(predicate)?.right() ?: default().left() }")
+  ReplaceWith("flatMap { if (predicate(it)) Right(it) else Left(default(it)) }")
 )
 public inline fun <A, B> Either<A, B>.filterOrElse(predicate: (B) -> Boolean, default: () -> A): Either<A, B> =
-  ensure(default, predicate)
+  flatMap { if (predicate(it)) Right(it) else Left(default()) }
```

Comment:
@bwanner, @dberg, you don't mind the deprecation? If you came across this, I am assuming you were relying on it in your codebase?

Reply:
Hi @nomisRev, yes we were using it but we've replaced it with a similar impl that has a specialized left value that we use to represent errors. And nope, tests and @bwanner caught the breaking change before any deploys were made.

Reply:
@nomisRev - on a related note we defined `rightIfNotNull`, `rightIfNull`, `leftIfNull` in the app itself since we found it useful for the expressiveness / readability.